GradedModulePresentationsForCAP/gap/GradedModulePresentations.gi

```diff
@@ -1108,6 +1632,273 @@ InstallGlobalFunction( ADD_GRADED_INTERNAL_HOM_ON_MORPHISMS_RIGHT,
+        
+        indeterminates := Indeterminates( S );
+        
+        degrees_of_indeterminates := List( indeterminates, Degree );
+        
+        random_matrix_between_free_left_or_right_presentations_func :=
+          function( left_or_right )
+            local func;
+            
+            func := function( degrees_1, degrees_2 )
+                      local mat;
+                      
+                      if left_or_right = "left" then
+                        
+                        mat := List( degrees_1,
+                             i -> List( degrees_2,
+                             j -> S!.random_homogeneous_element_func( i - j ) ) );
+                      
+                      else
+                        
+                        mat := List( degrees_1,
+                             i -> List( degrees_2,
+                             j -> S!.random_homogeneous_element_func( j - i ) ) );
+                      
+                      fi;
+                      
+                      if Length( degrees_1 ) * Length( degrees_2 ) = 0 then
+                        
+                        return HomalgZeroMatrix( Length( degrees_1 ), Length( degrees_2 ), S );
+                      
+                      else
+                        
+                        return HomalgMatrix( mat, S );
+                      
+                      fi;
+                    
+                    end;
+            
+            return func;
+          
+          end;
+        
+        random_matrix_for_left_or_right_presentation_func :=
+          function( random_matrix_between_free_left_or_right_presentations_func )
+            local func;
+            
+            func := function( m, n  )
+              local D, L, K;
+              
+              D := S!.random_degrees;
+              
+              L := List( [ 1 .. m ], i -> Random( D ) );
+              
+              K := List( [ 1 .. n ], i -> Random( D ) );
+              
+              if m * n = 0 then
+                
+                return [ HomalgZeroMatrix( m, n, S ), L, K ];
+              
+              else
+                
+                return [ random_matrix_between_free_left_or_right_presentations_func( L, K ), L, K ];
+              
+              fi;
+            
+            end;
+            
+            return func;
+          
+          end;
+        
+        if  ( Rank( DegreeGroup( S ) ) = 1 and
+                ( IsSubset( Set( [ 0, 1 ] ), Set( List( degrees_of_indeterminates, HomalgElementToInteger ) ) ) or
+                    IsSubset( Set( [ 0, -1 ] ), Set( List( degrees_of_indeterminates, HomalgElementToInteger ) ) ) ) ) then
+          
+          random_homogeneous_element_func :=
+            function( degree )
+              local mat, p, coeffs, r, l1, l2;
+              
+              if degree = Degree( Zero( S ) ) then
```

Comment:
This line unnecessarily prevents the generation of random elements that, simply by accident, have the same degree that was assigned to the zero element. See homalg-project/homalg_project#234

Reply:
This has been added as a quick fix to some issues appeared in a previous implementation of the random morphisms. In this commit
https://github.com/homalg-project/CAP_project/pull/239/commits/b0a4508125e494db673d67afe760137fea9eb211
I have updated the corresponding methods and hence removed this quick fix.
Thanks for  catching this :) I am much more happy with the current code.